File: ruby/bob/bob.rb
module BookKeeping
  VERSION = 1
end

class Bob

  def hey(str)
    return "Fine. Be that way!" if silent?(str) 
    return "Whoa, chill out!" if shouting?(str) 
    return "Sure." if asking?(str) 
    "Whatever."
  end

  def silent?(sentence)
    sentence.strip.empty?
  end

  def shouting?(sentence)
    sentence =~ /[A-Z]/ && sentence !~ /[a-z]/
  end

  def asking?(sentence)
    sentence.end_with?("?")
  end

end

File: python/bob/bob.py
to_strip = ['@','#','$','%','^','&','*','(',')','+','=','[',']','{','}','|',';',':','<','>']

def hey(phrase):
    if phrase.strip().endswith('?'):
        return is_question(phrase)
    elif blank(phrase):
        return "Fine. Be that way!"
    elif phrase.upper() == phrase and phrase.replace(' ','').replace(',','').isdigit() != True:
        return "Whoa, chill out!"
    else:
        return "Whatever."
        
def blank(phrase):
    if phrase.isspace() or len(phrase) == 0:
        return True
    
def is_question(phrase):
    for c in phrase:
        if c in to_strip:
            return "Sure."
    else:
        if phrase.upper() == phrase and phrase.strip('?').isdigit() != True:
            return "Calm down, I know what I'm doing!"
        else:
            return "Sure."




